R/qenv-join.R

```diff
@@ -7,102 +7,106 @@
-#'     env = list2env(list(mtcars1 = mtcars))
-#'   )
-#'   y <- qenv(
-#'     code = c(mtcars1 = "mtcars1 <- mtcars['wt']"),
-#'     env = list2env(list(mtcars1 = mtcars['wt']))
-#'   )
-#'   z <- join(x, y)
-#'   # Error message will occur
-#'   }
-#'   In this example, `mtcars1` object exists in both `x` and `y` objects but the content are not identical.\cr
-#'   `mtcars1` in the `x qenv` object has more columns than `mtcars1` in the `y qenv` object (only has one column).
-#'   \item `join()` will look for identical `@id` values in both `qenv` objects.
-#'   The index position of these `@id`s must be the same to determine the evaluation order.
-#'   Otherwise, `join()` will throw an error message.\cr\cr
-#'   Example:
-#'   \preformatted{
-#'   common_q <- qenv(code = "v <- 1", env = list2env(list(v = 1)))
-#'   x <- eval_code(
-#'     common_q,
-#'     "x <- v"
-#'   )
-#'   y <- eval_code(
-#'     common_q,
-#'     "y <- v"
-#'   )
-#'   z <- eval_code(
-#'     y,
-#'     "z <- v"
-#'   )
-#'   q <- join(x, y)
-#'   join_q <- join(q, z)
-#'   # Error message will occur
-#'
-#'   # Check the order of evaluation based on the id slot
-#'   shared_ids <- intersect(q@id, z@id)
-#'   match(shared_ids, q@id) # Output: 1 3
-#'   match(shared_ids, z@id) # Output: 1 2
-#'   }
-#'   The error occurs because the index position of identical `@id` between the two objects is not the same.
-#'   \item The usage of temporary variable in the code expression could cause `join()` to fail. \cr\cr
-#'   Example:
-#'   \preformatted{
-#'   common_q <- qenv()
-#'   x <- eval_code(
-#'     common_q,
-#'     "x <- numeric(0)
-#'      for (i in 1:2) {
-#'        x <- c(x, i)
-#'      }"
-#'   )
-#'   y <- eval_code(
-#'     common_q,
-#'     "y <- numeric(0)
-#'      for (i in 1:3) {
-#'        y <- c(y, i)
-#'      }"
-#'   )
-#'   q <- join(x,y)
-#'   # Error message will occur
-#'
-#'   # Check the value of temporary variable i in both objects
-#'   x@env$i # Output: 2
-#'   y@env$i # Output: 3
-#'   }
-#'   `join()` fails to provide a proper result because of the temporary variable `i` exists
-#'   in both objects but has different value.\cr
-#'   To fix this, we can set `i <- NULL` in the code expression for both objects.
-#'   \preformatted{
-#'   common_q <- qenv()
-#'   x <- eval_code(
-#'     common_q,
-#'     "x <- numeric(0)
-#'      for (i in 1:2) {
-#'        x <- c(x, i)
-#'      }
-#'      # dummy i variable to fix it
-#'      i <- NULL"
-#'   )
-#'   y <- eval_code(
-#'     common_q,
-#'     "y <- numeric(0)
-#'      for (i in 1:3) {
-#'        y <- c(y, i)
-#'      }
-#'      # dummy i variable to fix it
-#'      i <- NULL"
-#'   )
-#'   q <- join(x,y)
-#'   }
-#' }
+#' 1. Both `qenv` objects contain an object of the same name but are not identical.
+#'
+#'    Example:
+#'
+#'    ```r
+#'    x <- qenv(
+#'      code = c(mtcars1 = "mtcars1 <- mtcars"),
+#'      env = list2env(list(mtcars1 = mtcars))
+#'    )
```

Comment:
This code doesn't work for me:

![image](https://github.com/insightsengineering/teal.code/assets/8597300/1a57e8aa-7111-4732-9538-10d4d7297a2c)

Should this be updated using `within` or `eval_code` since we no longer recommend `new_qenv`?

Reply:
Something like this?

```r
x <- qenv()
x <- eval_code(x, "mtcars1 <- mtcars")
y <- qenv()
y <- eval_code(y, "mtcars1 <- mtcars['wt']")
z <- join(x, y)
```

![image](https://github.com/insightsengineering/teal.code/assets/8597300/7e295e6d-eb7b-42a3-ab3a-523142cbe534)

Reply:
These were not under @example tag, but were under documentation, so they got missed."